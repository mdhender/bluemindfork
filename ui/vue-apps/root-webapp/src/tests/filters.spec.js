import {
    ACTIONS,
    CRITERIA_TARGETS,
    CRITERIA_MATCHERS,
    MATCH_ALL,
    read,
    write
} from "../components/preferences/fields/customs/FilterRules/filterRules.js";

describe("filters", () => {
    test("read", () => {
        const criteriaString = `FROM:IS: Toto TO:IS: Tata SUBJECT:CONTAINS: Toto&Tata 
        BODY:DOESNOTCONTAIN: un gros mot X-My-Header:MATCHES: head-bang`;
        const rawFilter = {
            active: true,
            criteria: criteriaString,
            delete: false,
            deliver: "INBOX",
            discard: false,
            forward: { enabled: true, localCopy: false, emails: ["toto.test@bluemind.net"] },
            name: "N/A",
            read: true,
            star: false,
            terminal: true
        };
        const filter = read(rawFilter);
        expect(filter).toMatchSnapshot();
    });

    test("read - no criteria", () => {
        const rawFilter = {
            active: true,
            criteria: MATCH_ALL,
            delete: false,
            deliver: "INBOX",
            discard: false,
            forward: { enabled: false, localCopy: true, emails: ["toto.test@bluemind.net"] },
            name: "N/A",
            read: true,
            star: false,
            terminal: true
        };
        const filter = read(rawFilter);
        expect(filter).toMatchSnapshot();
    });

    test("write", () => {
        const filter = {
            actions: [
                { type: ACTIONS.DELETE.type, value: true },
                {
                    type: ACTIONS.FORWARD.type,
                    value: { localCopy: false, emails: ["toto.test@bluemind.net"] }
                }
            ],
            active: true,
            criteria: [
                {
                    target: { type: CRITERIA_TARGETS.BODY.type, name: "BODY" },
                    matcher: CRITERIA_MATCHERS.DOESNOTCONTAIN,
                    value: "un gros mot"
                },
                {
                    target: { type: CRITERIA_TARGETS.TO.type, name: "TO" },
                    matcher: CRITERIA_MATCHERS.IS,
                    value: "Tata"
                },
                {
                    target: { type: CRITERIA_TARGETS.HEADER.type, name: "X-Machin-Chouette" },
                    matcher: CRITERIA_MATCHERS.IS,
                    value: "Bidule"
                }
            ],
            exceptions: [],
            name: "MyFilter",
            terminal: true
        };
        const rawFilter = write(filter);
        expect(rawFilter).toMatchSnapshot();
    });

    test("write - no criteria", () => {
        const filter = {
            actions: [
                { type: ACTIONS.DELETE.type, value: true },
                {
                    type: ACTIONS.FORWARD.type,
                    value: { localCopy: false, emails: ["toto.test@bluemind.net"] }
                }
            ],
            active: true,
            criteria: [],
            exceptions: [],
            name: "MyFilter",
            terminal: true
        };
        const rawFilter = write(filter);
        expect(rawFilter).toMatchSnapshot();
    });
});
