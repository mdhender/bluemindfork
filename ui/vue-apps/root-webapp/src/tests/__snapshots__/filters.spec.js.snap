// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filters read - no criteria 1`] = `
Object {
  "actions": Array [
    Object {
      "type": "deliver",
      "value": "INBOX",
    },
    Object {
      "type": "forward",
      "value": Object {
        "emails": Array [
          "toto.test@bluemind.net",
        ],
        "enabled": false,
        "localCopy": true,
      },
    },
    Object {
      "type": "read",
      "value": true,
    },
  ],
  "active": true,
  "criteria": Array [],
  "exceptions": Array [],
  "name": "N/A",
  "terminal": true,
}
`;

exports[`filters read 1`] = `
Object {
  "actions": Array [
    Object {
      "type": "deliver",
      "value": "INBOX",
    },
    Object {
      "type": "forward",
      "value": Object {
        "emails": Array [
          "toto.test@bluemind.net",
        ],
        "enabled": true,
        "localCopy": false,
      },
    },
    Object {
      "type": "read",
      "value": true,
    },
  ],
  "active": true,
  "criteria": Array [
    Object {
      "matcher": "IS",
      "target": Object {
        "name": "FROM",
        "type": "FROM",
      },
      "value": "Toto",
    },
    Object {
      "matcher": "IS",
      "target": Object {
        "name": "TO",
        "type": "TO",
      },
      "value": "Tata",
    },
    Object {
      "matcher": "CONTAINS",
      "target": Object {
        "name": "SUBJECT",
        "type": "SUBJECT",
      },
      "value": "Toto&Tata",
    },
    Object {
      "matcher": "DOESNOTCONTAIN",
      "target": Object {
        "name": "BODY",
        "type": "BODY",
      },
      "value": "un gros mot",
    },
    Object {
      "matcher": "MATCHES",
      "target": Object {
        "name": "X-My-Header",
        "type": "HEADER",
      },
      "value": "head-bang",
    },
  ],
  "exceptions": Array [],
  "name": "N/A",
  "terminal": true,
}
`;

exports[`filters write - no criteria 1`] = `
Object {
  "active": true,
  "criteria": "MATCHALL",
  "delete": true,
  "deliver": undefined,
  "discard": false,
  "forward": Object {
    "emails": Array [
      "toto.test@bluemind.net",
    ],
    "enabled": true,
    "localCopy": false,
  },
  "name": "MyFilter",
  "read": false,
  "star": false,
  "stop": true,
}
`;

exports[`filters write 1`] = `
Object {
  "active": true,
  "criteria": "BODY:DOESNOTCONTAIN: un gros mot
TO:IS: Tata
X-Machin-Chouette:IS: Bidule
",
  "delete": true,
  "deliver": undefined,
  "discard": false,
  "forward": Object {
    "emails": Array [
      "toto.test@bluemind.net",
    ],
    "enabled": true,
    "localCopy": false,
  },
  "name": "MyFilter",
  "read": false,
  "star": false,
  "stop": true,
}
`;
